app/uk/gov/hmrc/nrs/retrieval/controllers/testonly/StrideAuthAction.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nrs.retrieval.controllers.testonly
+
+import com.google.inject.Inject
+import play.api.mvc.Results._
+import play.api.mvc._
+import uk.gov.hmrc.auth.core.AuthProvider.PrivilegedApplication
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.allEnrolments
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthProviders, AuthorisedFunctions, Enrolments}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton()
+class StrideAuthAction @Inject()(override val authConnector: AuthConnector, mcc: MessagesControllerComponents)(
+  implicit ec: ExecutionContext)
+    extends ActionBuilder[Request, AnyContent] with AuthorisedFunctions {
+
+  override val parser: BodyParser[AnyContent] = mcc.parsers.defaultBodyParser
+  override protected val executionContext: ExecutionContext = ec
+
+  override def invokeBlock[A](request: Request[A], block: Request[A] => Future[Result]): Future[Result] = {
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
```

Comment:
Invokes the block only if the user is authenticated via `stride` and authorised by having the correct enrolment.

Distinguishes between `401 UNAUTHORISED` for requests that are not `stride` authenticated and `403 FORBIDDEN` for requests which are `stride` authenticated but lack the correct enrolment.

The role is hard-coded whereas it is in config in the frontend. This is a more minimal change here so is preferred for a spike in any case. That said I can't see any advantage in having the role in config anyway...